services/apps/src/apps/clients/metadata.clj

```diff
@@ -0,0 +1,66 @@
+(ns apps.clients.metadata
+  (:use [kameleon.uuids :only [uuidify]])
+  (:require [cemerick.url :as curl]
+            [cheshire.core :as json]
+            [clj-http.client :as http]
+            [apps.util.config :as config]
+            [apps.util.service :as util]))
+
+(defn- metadata-url-encoded
+  [& components]
+  (str (apply curl/url (config/metadata-base) (map curl/url-encode components))))
+
```

Comment:
`get-options` and `post-options` make the rest of the code in this namespace a lot more concise. 👍